### PR TITLE
feat: Improve organization deletions UX

### LINKED ISSUE #6906: Show details when pending deletion
Specifically the main priority is organizations. If you queue an organization for deletion we should show a page which details that fact, the ETA of when itll begin, and the link to restore it (if you have the permissions).

### PR DESCRIPTION
**Pull request by dcramer**
- Override most organization pages to show deletion notification (React)
- Fix 'status.id' to be lowercase in API response

Fixes GH-6906

### CODE REVIEW

File: tests/js/setup.js [hunk 1]
```diff
@@ -159,13 +158,17 @@ window.TestStubs = {
...
+        name: 'active',
+      },
       features: [],
       onboardingTasks: [],
       teams: [],
       ...params,
     };
   },
-  Repository: (...params) => {
+  Repository: params => {
```
Comment (dcramer) [inline tests/js/setup.js]: when i implemented this in Zeus I noticed the incorrect use of ...params expansion here (which wont convert to dict values later)

**Comment by mattrobenolt** [comment]
I mean, the python bit is just `.lower()`. Hopefully @billyvg can check the all this JS. :)

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by dcramer** [comment]
Im assuming the Percy changes are unrelated as I dont know how I would have caused that

**Comment by dcramer** [comment]
screen shot 2018-01-08 at 2 36 44 pm

File: src/sentry/static/sentry/app/views/organizationDetails.jsx [hunk 2]
```diff
@@ -1,20 +1,146 @@
-import React from 'react';
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
 
+import AlertActions from '../actions/alertActions';
+import {Client} from '../api';
 import OrganizationContext from './organizationContext';
-
+import NarryLayout from '../components/narrowLayout';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDetails.jsx]: what's a Narry

File: src/sentry/static/sentry/app/views/organizationDetails.jsx [hunk 3]
```diff
@@ -1,20 +1,146 @@
...
+
+        {access.has('org:admin') ? (
+          <div>
+            <p>
+              {t(
+                'Would you like to cancel this process and restore the organization back to the original state?'
+              )}
+            </p>
+            <p>
+              <button type="submit" className="btn btn-primary" onClick={this.onRestore}>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDetails.jsx]: use `<Button priority="primary">`

**Review by billyvg — APPROVED** [review]

**Comment by billyvg** [comment]
Weird, percy started rendering our footer in mobile layouts?

image

**Comment by dcramer** [comment]
lol it literally didnt even let you submit the form before gj testing

File: src/sentry/static/sentry/app/views/organizationDetails.jsx (hunk 2 above)
Comment (dcramer) [reply src/sentry/static/sentry/app/views/organizationDetails.jsx]: i blame waking up at 4am

**Pull request merged**